code/game/objects/structures/kitchen_spike.dm

```diff
@@ -40,44 +51,72 @@
 		icon_state = "spike_[H.species.name]"
 	else
 		return FALSE
-
+	victim.loc = src
 	victim_name = victim.name
+	occupant = victim
 	occupied = TRUE
 	meat = 5
 	return TRUE
 
 /obj/structure/kitchenspike/attack_hand(mob/living/carbon/human/user)
 	if(..() || !occupied)
 		return
-	meat--
-	new meat_type(get_turf(src))
-	if(meat > 1)
-		to_chat(user, "You remove some meat from \the [victim_name].")
-	else if(meat == 1)
-		to_chat(user, "You remove the last piece of meat from \the [victim_name]!")
-		icon_state = initial(icon_state)
-		occupied = 0
-	if(meat_type == user.species.meat_type)
-		user.sanity.changeLevel(-(15*((user.nutrition ? user.nutrition : 1)/user.max_nutrition))) // The more hungry the less sanity damage.
-		to_chat(user, SPAN_NOTICE("You feel your [user.species.name]ity dismantling as you cut a slab off \the [src]")) // Human-ity , Monkey-ity , Slime-Ity
+	to_chat(user, "You start to remove [victim_name] from \the [src].")	
+	if(!do_after(user, 40))
+		return 0
+	occupant.loc = (get_turf(src))
+	occupied = FALSE
+	meat = 0
+	meat_type = initial(meat_type)
+	to_chat(user, "You remove [victim_name] from \the [src].")	
+	icon_state = initial(icon_state)
 
-
-/obj/structure/kitchenspike/attackby(obj/item/I, mob/user)
-	var/list/usable_qualities = list(QUALITY_BOLT_TURNING)
+/obj/structure/kitchenspike/attackby(obj/item/I, mob/living/carbon/human/user)
+	var/list/usable_qualities = list(QUALITY_BOLT_TURNING, QUALITY_CUTTING)
 	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+	if(tearing)
+		return
 
-	if (tool_type == QUALITY_BOLT_TURNING)
+	if(tool_type == QUALITY_CUTTING)
+		if(!occupied)
+			return
+		tearing = TRUE
+		if(do_after(user, 20))
+			tearing = FALSE
+			var/cut_damage
+			var/organ_damaged
+			organ_damaged = pickweight(list(BP_HEAD = 0.1, BP_GROIN = 0.2, BP_R_ARM = 0.2, BP_L_ARM = 0.2, BP_R_LEG = 0.2, BP_L_LEG = 0.2))
+			cut_damage = rand(15, 25)
+			occupant.apply_damage(cut_damage, BRUTE, organ_damaged)
+			playsound(src, 'sound/weapons/bladeslice.ogg', 50, 1)
+			if(meat < 1)
+				to_chat(user, "There is no more meat on \the [victim_name].")
+				return
+			meat--
+			new meat_type(get_turf(src))
+			if(meat > 1)
+				to_chat(user, "You remove some meat from \the [victim_name].")
+			else if(meat == 1)
+				to_chat(user, "You remove the last piece of meat from \the [victim_name]!")
```

Comment:
You need to either:
* put that before the `meat--`
* put `>= 1` and `== 0` instead.